relayer/crypto/merkle/simplified_mmr_proof_test.go

```diff
@@ -0,0 +1,110 @@
+
+
+
+
+func Test_SimplifiedMMRProof(t *testing.T) {
+	var mmrProofs []types.H256
+	mmrProofItems := []string{
+		"ee29f67540b840ec61825261182b7a8e73d1a1e548271b18e619ea87e0d1963e",
+		"a244ae2c1d1ad04bbb307ce418d84980aa656c6818516f019364df29c1a37224",
+		"1acbd16b3dab0a66914fbff7cbaa15f2c4a37db6d3552ff1a466cc51cb2cbb8b",
+		"58059c5ee4970ecc8491209c64dae9b5b850dce7e983bcf982079a6423d74ef4",
+		"6aafdb1600e3233eb7cf374a438b64a11e8387b9cd0145557634c013e670c9c8",
+	}
+	for _, item := range mmrProofItems {
+		out, err := hex.DecodeString(item)
+		assert.NoError(t, err)
+		mmrProofs = append(mmrProofs, types.NewH256(out))
+	}
+	simplifiedMMRProof, err := ConvertToSimplifiedMMRProof(types.H256{}, 70, types.MMRLeaf{}, 73, mmrProofs)
+	assert.NoError(t, err)
+	fixture := convertSimplifiedProofToFixture(simplifiedMMRProof)
+	prettyOut, err := json.MarshalIndent(fixture, "", "\t")
+	assert.NoError(t, err)
+	fmt.Println(string(prettyOut))
+
+	var testData []SimplifiedProofTestData
+	err = json.Unmarshal([]byte(TestData), &testData)
+	if err != nil {
+		t.Fail()
+	}
+	for i := 0; i < len(testData); i++ {
+		fmt.Printf("Testing for: LeafIndex; %d, LeafCount: %d\n", testData[i].LeafIndex, testData[i].LeafCount)
+		simplifiedProof, err := ConvertToSimplifiedMMRProof(testData[i].LeafHash, testData[i].LeafIndex, types.MMRLeaf{}, testData[i].LeafCount, testData[i].MMRProof)
+		assert.NoError(t, err)
+		assert.Equal(t, testData[i].ReferenceSimplifiedProof.MMRRestOfThePeaks, simplifiedProof.MMRRestOfThePeaks)
+		assert.Equal(t, testData[i].ReferenceSimplifiedProof.MMRRightBaggedPeak, simplifiedProof.MMRRightBaggedPeak)
+		assert.Equal(t, testData[i].ReferenceSimplifiedProof.MerkleProofOrder, simplifiedProof.MerkleProofOrder)
+		assert.Equal(t, testData[i].ReferenceSimplifiedProof.MerkleProofItems, simplifiedProof.MerkleProofItems)
+
+		fmt.Printf("\nRoot: %#x\n", testData[i].ReferenceMMRRoot)
+		fmt.Printf("Leaf Node hash: %#x\n", testData[i].LeafHash)
+
+		// Printing solidity fixture
+		fixture := convertSimplifiedProofToFixture(simplifiedProof)
+		prettyOut, err := json.MarshalIndent(fixture, "", "\t")
+		assert.NoError(t, err)
+		fmt.Println(string(prettyOut))
+	}
+}
```

Comment:
For better test coverage, could you please add the following two tests? 

1. Programmatically generate a 15 leaf MMR and then create and validate a proof for each of the 15 leaves, as well as validate proof failure with a leaf that's not in the MMR. 
2. Verify that the code works on a 1 leaf MMR, as well as validate proof failure with a leaf that's not in the MMR.

Reply:
@philipstanislaus 

1. The go code does not contain the logic to create and validate a proof for each of the 15 leaves. This test exists on the Rust side.
2. Same as above.